esp-hal/src/i2s/master/camera.rs

```diff
@@ -0,0 +1,404 @@
+
+        // Default these signals to high in case user doesn't provide them.
+        I2S::v_sync_signal().connect_to(Level::High);
+        I2S::h_sync_signal().connect_to(Level::High);
+        I2S::h_enable_signal().connect_to(Level::High);
+
+        let rx_channel = ChannelRx::new(channel.map(|ch| ch.degrade()));
+
+        Self {
+            _i2s: i2s,
+            rx_channel,
+        }
+    }
+}
+
+impl<I2S> Camera<'_, I2S>
+where
+    I2S: RegisterAccess + ExtendedSignals,
+{
+    /// Configures the data pins for the camera interface.
+    #[allow(clippy::too_many_arguments)]
+    pub fn with_data_pins<
+        D0: InputPin,
+        D1: InputPin,
+        D2: InputPin,
+        D3: InputPin,
+        D4: InputPin,
+        D5: InputPin,
+        D6: InputPin,
+        D7: InputPin,
+    >(
+        self,
+        d0: impl Peripheral<P = D0>,
+        d1: impl Peripheral<P = D1>,
+        d2: impl Peripheral<P = D2>,
+        d3: impl Peripheral<P = D3>,
+        d4: impl Peripheral<P = D4>,
+        d5: impl Peripheral<P = D5>,
+        d6: impl Peripheral<P = D6>,
+        d7: impl Peripheral<P = D7>,
+    ) -> Self {
+        crate::into_mapped_ref!(d0);
+        crate::into_mapped_ref!(d1);
+        crate::into_mapped_ref!(d2);
+        crate::into_mapped_ref!(d3);
+        crate::into_mapped_ref!(d4);
+        crate::into_mapped_ref!(d5);
+        crate::into_mapped_ref!(d6);
+        crate::into_mapped_ref!(d7);
+
+        let pairs = [
+            (d0, I2S::din0_signal()),
+            (d1, I2S::din1_signal()),
+            (d2, I2S::din2_signal()),
+            (d3, I2S::din3_signal()),
+            (d4, I2S::din4_signal()),
+            (d5, I2S::din5_signal()),
+            (d6, I2S::din6_signal()),
+            (d7, I2S::din7_signal()),
+        ];
+
+        for (pin, signal) in pairs {
+            pin.init_input(Pull::None);
+            signal.connect_to(pin);
+        }
+
+        self
+    }
+
+    /// Configures the pixel clock pin for the camera interface.
+    pub fn with_ws<PIN: InputPin>(self, pin: impl Peripheral<P = PIN>) -> Self {
+        crate::into_mapped_ref!(pin);
+        pin.init_input(Pull::None);
+        I2S::ws_in_signal().connect_to(pin);
+        self
+    }
+
+    /// Configures the vertical sync (VSYNC) pin for the camera interface.
+    pub fn with_vsync<PIN: InputPin>(self, pin: impl Peripheral<P = PIN>) -> Self {
+        crate::into_mapped_ref!(pin);
+        pin.init_input(Pull::None);
+        I2S::v_sync_signal().connect_to(pin);
+        self
+    }
+
+    /// Configures the horizontal sync (HSYNC) pin for the camera interface.
+    pub fn with_hsync<PIN: InputPin>(self, pin: impl Peripheral<P = PIN>) -> Self {
+        crate::into_mapped_ref!(pin);
+        pin.init_input(Pull::None);
+        I2S::h_sync_signal().connect_to(pin);
+        self
+    }
+
+    /// Configures the vertical sync enable pin for the camera interface.
```

Comment:
typo

Reply:
i don't see it? 🙈

Reply:
`with_h_enable` vs "vertical sync enable pin"